operators/audio/sampling.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <vector>
+#include <cmath>
+#include <complex>
+#include "narrow.h"
+
+// https://en.wikipedia.org/wiki/Butterworth_filter
+class ButterworthLowpass {
+ public:
+  ButterworthLowpass(float sample_rate, float cutoff_frequency)
+      : x_prev_(0.0f), y_prev_(0.0f) {
+    float RC = 1.0f / (2.0f * 3.14159265359f * cutoff_frequency);
+    float dt = 1.0f / sample_rate;
+    float alpha = dt / (RC + dt);
+    a0_ = alpha;
+    a1_ = alpha;
+    b1_ = 1 - alpha;
+  }
+
+  float Process(float input) {
+    float output = a0_ * input + a1_ * x_prev_ - b1_ * y_prev_;
+    x_prev_ = input;
+    y_prev_ = output;
+    return output;
+  }
+
+  std::vector<float> Process(const std::vector<float>& inputSignal) {
+    std::vector<float> outputSignal(inputSignal.size());
+    for (size_t i = 0; i < inputSignal.size(); ++i) {
+      outputSignal[i] = Process(inputSignal[i]);
+    }
+    return outputSignal;
+  }
+
+ private:
+  float x_prev_, y_prev_;
+  float a0_, a1_, b1_;
+};
+
+// https://en.wikipedia.org/wiki/Kaiser_window
```

Comment:
should probably replace with more credited source? i found this Stanford site: https://ccrma.stanford.edu/~jos/sasp/Kaiser_Window.html

Reply:
yes, this one looks very concise, thanks!